Symfony/CS/Fixer/Contrib/MethodArgumentDefaultValueFixer.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\FixerInterface;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Mark Scherer
+ * @author Lucas Manzke <lmanzke@outlook.com>
+ */
+final class MethodArgumentDefaultValueFixer extends AbstractFixer
+{
+    private $argumentBoundaryTokens = array('(', ',', ')', ';', '{', '}');
+    private $variableOrTerminatorTokens = array(array(T_VARIABLE), ')', ';', '{', '}');
+    private $argumentTerminatorTokens = array(',', ')', ';', '{');
+    private $defaultValueTokens = array('=', ')', ';', '{');
+    private $immediateDefaultValueTokens = array('=', ')', ',', ';', '{');
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'In method arguments there must not be arguments with default values before non-default ones.';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $tokens = Tokens::fromCode($content);
+        for ($i = 0, $l = $tokens->count(); $i < $l; ++$i) {
+            if ($tokens[$i]->isGivenKind(T_FUNCTION)) {
+                $this->fixFunctionDefinition($tokens, $i);
+            }
+        }
+
+        return $tokens->generateCode();
+    }
+
+    /**
+     * @param Tokens $tokens
```

Comment:
I bet you are talking here about feature of PHPStorm, but looking on pure PHPDoc that notation would be wrong, because you cant pass `Token[]` to the method!

Typehinting of `$tokens[$nextRelevantIndex]` should be done thanks to https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/1.11/Symfony/CS/Tokenizer/Tokens.php#L23